packages/blob-storage/internal/logic/callbacks.go

```diff
@@ -0,0 +1,188 @@
+// callbacks.go
+
+package logic
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"log/slog"
+	"math/big"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/taikoxyz/taiko-mono/packages/blob-storage/internal/taikol1"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type Response struct {
+	Data []struct {
+		Index            string `json:"index"`
+		Blob             string `json:"blob"`
+		KzgCommitment    string `json:"kzg_commitment"`
+		KzgCommitmentHex []byte `json:"-"`
+	} `json:"data"`
+}
+
+// Callback functions
+
+// BlockProposedCallback is a callback for the "BlockProposed" event.
+func BlockProposedCallback(rpcURL, beaconURL, networkName string, log types.Log) {
+
+	contractAbi, err := abi.JSON(strings.NewReader(taikol1.TaikoL1ABI))
+	if err != nil {
+		slog.Error("Could not initiate reader")
```

Comment:
if we encounter an error here, we don't return, only log,  so the rest of the function will continue. also, we aren't logging what error occurred